meta-mbl-distro/scripts/lib/mbl/update/payload.py

```diff
@@ -9,20 +9,19 @@
 import subprocess
 import tempfile
 
-import mbl.update.appsimage as appsimage
-import mbl.update.bootimage as bootimage
-import mbl.update.rootfsimage as rootfsimage
-import mbl.update.swdesc as swdesc
+import mbl.update.payloadbuilder as payloadbuilder
 import mbl.update.testinfo as testinfo
-import mbl.update.wksbootloaderslotimage as wksbootloaderslotimage
 import mbl.util.tinfoilutil as tutil
 
+CURRENT_PAYLOAD_FORMAT_VERSION = 3
+
 
 class UpdatePayload:
     """Class for creating update payloads and metadata."""
 
     def __init__(
         self,
+        payload_format_version,
         tinfoil,
         bootloader_components=[],
```

Comment:
I should've caught this in the last review, we shouldn't be using 'mutable default arguments' as they behave especially weirdly in python. The list object is created at parse time, instead of a new list being created on each instantiation of the object, so if it's mutated in any way it will be mutated for all instances of the class. Might not practically make a difference in this case as this class is a singleton anyway, but it's bad practice to do this in general. We should use `None` as the default value here (and for `apps`) instead of an empty list.

Reply:
Okay, will address in the code. I wonder if I can effectively use tuples to make this go away...

Reply:
https://github.com/ARMmbed/meta-mbl/pull/781/commits/200033857e84ab4f64b29f85dcdb417c57d96c35